rust/src/sdk/api.rs

```diff
@@ -102,21 +103,33 @@ pub struct FFIClient {
 #[no_mangle]
 /// Creates a new [`Client`] within an asynchronous runtime.
 ///
+/// Takes a network `address` to the coordinator to which the [`Client`] will try to connect to.
+///
 /// Takes a `period` in seconds after which the [`Client`] will try to poll the coordinator for new
 /// broadcasted FL round data again.
 ///
 /// # Errors
-/// Returns a null pointer in case of invalid arguments or if the initialization of the runtime
-/// or the client fails. The following must hold true for arguments to be valid:
-/// - `period` > 0
+/// Ignores null pointer `address`es and zero `period`s and returns a null pointer immediately.
+///
+/// Returns a null pointer if the initialization of the runtime or the client fails.
 ///
 /// # Safety
-/// The method depends on the safety of the `callback` and on the consistent definition and layout
-/// of its `input` across the FFI-boundary.
-pub unsafe extern "C" fn new_client(period: c_ulonglong) -> *mut FFIClient {
-    if period == 0 {
+/// The method dereferences from the raw pointer arguments. Therefore, the behavior of the method is
+/// undefined if the arguments don't point to valid objects.
+pub unsafe extern "C" fn new_client(address: *const c_char, period: c_ulonglong) -> *mut FFIClient {
+    if address.is_null() || period == 0 {
         return ptr::null_mut() as *mut FFIClient;
     }
+    let address = if let Ok(address) = unsafe {
```

Comment:
Just for my understanding.
`address` points to the address of c_char. So it does not copy the value of `address` but since `address` is `*const` it is safe because the value can't be dropped in the c-code. Is that right?

Reply:
that's right, memory management for the original string object is a task for the original owner of the string. to summarize what this is doing:
- input a char pointer to a string
- do some utf-8 and null validation
- create a rust reference to the original string object

`CStr` is closely related to `&str`, whereas the owned version are the `CString` which is closely related to `String`. it is important that the `CStr` reference (much like the `&str`)  is only valid as long as the original string object exists. this approach should be fine for a short read at the time of calling. if we need to store the address for later use, we should make a proper copy via `CString`.